src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -184,43 +186,10 @@ public class SettingsController implements Controller<Node> {
 
   private ChangeListener<Theme> selectedThemeChangeListener;
   private ChangeListener<Theme> currentThemeChangeListener;
-
-  public SettingsController(ApplicationContext applicationContext, UserService userService, PreferencesService preferencesService, UiService uiService,
-                            I18n i18n, EventBus eventBus, NotificationService notificationService,
-                            PlatformService platformService, ClientProperties clientProperties,
-                            ClientUpdateService clientUpdateService, GameService gameService, TaskService taskService, CoturnService coturnService, IceServerMapper iceServerMapper) {
-    this.applicationContext = applicationContext;
-    this.userService = userService;
-    this.preferencesService = preferencesService;
-    this.uiService = uiService;
-    this.i18n = i18n;
-    this.eventBus = eventBus;
-    this.notificationService = notificationService;
-    this.platformService = platformService;
-    this.clientProperties = clientProperties;
-    this.clientUpdateService = clientUpdateService;
-    this.gameService = gameService;
-    this.taskService = taskService;
-    this.coturnService = coturnService;
-    this.iceServerMapper = iceServerMapper;
-
-    availableLanguagesListener = observable -> {
-      LocalizationPrefs localization = preferencesService.getPreferences().getLocalization();
-      Locale currentLocale = localization.getLanguage();
-      List<Node> nodes = i18n.getAvailableLanguages().stream()
-          .map(locale -> {
-            LanguageItemController controller = uiService.loadFxml("theme/settings/language_item.fxml");
-            controller.setLocale(locale);
-            controller.setOnSelectedListener(this::onLanguageSelected);
-            controller.setSelected(locale.equals(currentLocale));
-            return controller.getRoot();
-          })
-          .collect(Collectors.toList());
-      languagesContainer.getChildren().setAll(nodes);
-    };
-  }
+  private InvalidationListener availableLanguagesListener;
```

Comment:
Small nit is just that even though it doesn't allow for a nice lombok constructor this should probably still be final but not gonna hold up this PR because of that